File: ima-frontend-react/src/services/StudentService.js
import securedAxiosInstance from "./SecuredAxiosInstance";

const REST_API_STUDENTS = '/students';

export const listStudents = () => securedAxiosInstance.get(`${REST_API_STUDENTS}`);

export const getStudent = (studentId) => securedAxiosInstance.get(`${REST_API_STUDENTS}/${studentId}`);

export const deleteStudent = (studentId) => securedAxiosInstance.delete(`${REST_API_STUDENTS}/${studentId}`);

export const createStudent = (userId, studentRequest) =>
    securedAxiosInstance.post(
        REST_API_STUDENTS,
        studentRequest,
        {
            params: { userId }
        }
    );

export const updateStudent = (studentId, studentRequest) => securedAxiosInstance.put(`${REST_API_STUDENTS}/${studentId}`, studentRequest);

File: ima-backend/src/main/java/com/techsolution/ima_backend/services/StudentService.java
package com.techsolution.ima_backend.services;

import com.techsolution.ima_backend.dtos.request.StudentRequest;
import com.techsolution.ima_backend.dtos.response.StudentResponse;


import java.util.List;

public interface StudentService {
    StudentResponse createStudent(Long userId, StudentRequest studentRequest);
    List<StudentResponse> getAllStudents();
    StudentResponse getStudentById(Long studentId);
    StudentResponse updateStudent(Long studentId, StudentRequest studentRequest);
    void deleteStudent(Long studentId);
}
